tests/sagemaker/test_sagemaker_deployment_client.py

```diff
@@ -424,19 +424,22 @@ def test_attempting_to_deploy_in_asynchronous_mode_without_archiving_throws_exce
 
 @mock_sagemaker_aws_services
 def test_create_deployment_create_sagemaker_and_s3_resources_with_expected_tags_from_local(
-    pretrained_model, sagemaker_client, sagemaker_deployment_client
+    pretrained_model, sagemaker_client, sagemaker_deployment_client, monkeypatch
 ):
     expected_tags = [{"Key": "key1", "Value": "value1"}, {"Key": "key2", "Value": "value2"}]
 
     name = "test-app"
-    with mock.patch.dict(os.environ, {}, clear=True):
-        sagemaker_deployment_client.create_deployment(
-            name=name,
-            model_uri=pretrained_model.model_uri,
-            config=dict(
-                tags={"key1": "value1", "key2": "value2"},
-            ),
-        )
+    monkeypatch.delenv("AWS_ACCESS_KEY_ID", raising=False)
+    monkeypatch.delenv("AWS_SECRET_ACCESS_KEY", raising=False)
+    monkeypatch.delenv("AWS_SESSION_TOKEN", raising=False)
+    monkeypatch.delenv("AWS_DEFAULT_REGION", raising=False)
```

Comment:
This looks more complicated 🤔

Reply:
@serena-ruan Yes, but more clear. `with mock.patch.dict(os.environ, {}, clear=True):` doesn't tell us which variables need to be erased.

Reply:
`with mock.patch.dict(os.environ, {}, clear=True):` removes unrelated environment variables.

Reply:
Fair enough